src/Microsoft.AspNetCore.Session/DistributedSession.cs

```diff
@@ -210,6 +261,12 @@ private void Deserialize(Stream content)
                 int dataLength = DeserializeNumFrom4Bytes(content);
                 _store[key] = ReadBytes(content, dataLength);
             }
+
+            if (_logger.IsEnabled(LogLevel.Debug))
+            {
+                _sessionId = new Guid(_sessionIdBytes).ToString();
```

Comment:
Ah got it. Missed the part where you set the session id bytes. (Was initially wondering if when log is not enabled and users try to get an Id, would they get a new Id than the stored one, but this is not the case)